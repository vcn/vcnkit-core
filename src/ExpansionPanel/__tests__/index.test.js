import React from 'react';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import * as Panel from '../';

afterEach(cleanup);

describe('ExpansionPanel', () => {
    test('Panel renders', () => {
        const { asFragment } = render(
            <Panel.Panel>
                <Panel.Header>Header</Panel.Header>
                <Panel.Content>Content</Panel.Content>
            </Panel.Panel>           
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('Panel renders with custom icon', () => {
        const CustomIcon = () => <p>CustomIcon</p>;

        const { getByText, asFragment } = render(
            <Panel.Panel>
                <Panel.Header icon={ <CustomIcon /> }>Header</Panel.Header>
                <Panel.Content>Content</Panel.Content>
            </Panel.Panel>       
        );

        expect(getByText('CustomIcon')).toHaveTextContent('CustomIcon');
        expect(asFragment()).toMatchSnapshot();
    });

    test('Panel expands and collapses with local state', async () => {
        const { getByText } = render(
            <Panel.Panel>
                <Panel.Header>Header</Panel.Header>
                <Panel.Content>
                    <p>Child</p>
                </Panel.Content>
            </Panel.Panel>           
        );

        fireEvent.click(getByText('Header'));

        const node = await waitForElement(() => getByText('Child'));
        expect(node).toHaveTextContent('Child');
    });

    test('Panel expands and collapsed with prop', () => {
        const { queryByText, rerender } = render(
            <Panel.Panel expanded={ false }>
                <Panel.Header>Header</Panel.Header>
                <Panel.Content>Content</Panel.Content>
            </Panel.Panel> 
        );

        expect(queryByText('Content')).not.toBeInTheDocument();

        rerender(
            <Panel.Panel expanded>
                <Panel.Header>Header</Panel.Header>
                <Panel.Content>Content</Panel.Content>
            </Panel.Panel> 
        );

        expect(queryByText('Content')).toBeInTheDocument();
    });
});