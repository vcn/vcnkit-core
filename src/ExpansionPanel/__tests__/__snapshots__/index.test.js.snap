// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansionPanel Panel renders 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0;
  margin: 0;
  -webkit-transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1);
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1);
  box-shadow: 0rem 0rem 0.125rem 0rem hsla(0, 0%, 0%, 0.14),0rem 0.125rem 0.125rem 0rem hsla(0, 0%, 0%, 0.12),0rem 0.0625rem 0.1875rem 0rem hsla(0, 0%, 0%, 0.2);
  background: #FFFFFF;
}

.c0:first-of-type {
  margin-top: 0;
  border-radius: .125rem .125rem 0 0;
}

.c0:last-of-type {
  border-radius: 0 0 .125rem .125rem;
}

.c4 {
  -webkit-transition: .2s ease-in-out;
  transition: .2s ease-in-out;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  height: 1.5rem;
  width: 1.5rem;
  fill: hsla(0, 0%, 0%, .57);
  display: block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: content-box;
  padding: 0rem 1.5rem;
  min-height: 3.25rem;
  -webkit-transition: padding 0.2s;
  transition: padding 0.2s;
  cursor: pointer;
  font-family: Roboto,sans-serif;
  color: hsla(0, 0%, 0%, .87);
  font-size: .9375rem;
  font-weight: 400;
  line-height: 1.33333333;
  letter-spacing: -0.008em;
}

.c1:focus {
  outline: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c3 {
  padding-left: 1.5rem;
}

@media all and (-ms-high-contrast: none),(-ms-high-contrast: active) {
  .c0 {
    border-bottom: .0625rem solid #EFEFEF;
  }
}

@media all and (-ms-high-contrast: none),(-ms-high-contrast: active) {
  .c1 {
    height: 3.25rem;
  }
}

<div
    class="c0"
  >
    <header
      aria-expanded="false"
      class="c1"
      role="button"
    >
      <div
        class="c2"
      >
        Header
      </div>
      <div
        class="c3"
      >
        <svg
          class="c4"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </div>
    </header>
    <div
      style="height: 1px;"
    />
  </div>
</DocumentFragment>
`;

exports[`ExpansionPanel Panel renders with custom icon 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0;
  margin: 0;
  -webkit-transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1);
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1);
  box-shadow: 0rem 0rem 0.125rem 0rem hsla(0, 0%, 0%, 0.14),0rem 0.125rem 0.125rem 0rem hsla(0, 0%, 0%, 0.12),0rem 0.0625rem 0.1875rem 0rem hsla(0, 0%, 0%, 0.2);
  background: #FFFFFF;
}

.c0:first-of-type {
  margin-top: 0;
  border-radius: .125rem .125rem 0 0;
}

.c0:last-of-type {
  border-radius: 0 0 .125rem .125rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: content-box;
  padding: 0rem 1.5rem;
  min-height: 3.25rem;
  -webkit-transition: padding 0.2s;
  transition: padding 0.2s;
  cursor: pointer;
  font-family: Roboto,sans-serif;
  color: hsla(0, 0%, 0%, .87);
  font-size: .9375rem;
  font-weight: 400;
  line-height: 1.33333333;
  letter-spacing: -0.008em;
}

.c1:focus {
  outline: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c3 {
  padding-left: 1.5rem;
}

@media all and (-ms-high-contrast: none),(-ms-high-contrast: active) {
  .c0 {
    border-bottom: .0625rem solid #EFEFEF;
  }
}

@media all and (-ms-high-contrast: none),(-ms-high-contrast: active) {
  .c1 {
    height: 3.25rem;
  }
}

<div
    class="c0"
  >
    <header
      aria-expanded="false"
      class="c1"
      role="button"
    >
      <div
        class="c2"
      >
        Header
      </div>
      <div
        class="c3"
      >
        <p>
          CustomIcon
        </p>
      </div>
    </header>
    <div
      style="height: 1px;"
    />
  </div>
</DocumentFragment>
`;
